woocommerce/woocommerce-gateway-payfast
# upkeep/222: modernize developer experience

### All Submissions:



---

### Changes proposed in this Pull Request:


- Node is updated to v20.
- NPM is updated to v10.
- Node and NPM changes are reflected in the Github actions workflows.
- Build assets are generated by [wordpress/scripts](https://www.npmjs.com/package/@wordpress/scripts) only.
- Entry point for all assets are moved under `src/`

Closes #222 

### Steps to test the changes in this Pull Request:



1. Pull the branch.
2. Run `nvm use` or `nvm install` depending on the situation.
3. Verify node and NPM versions by running `node -v && npm -v`.
4. `rm -rf node_modules/`.
5. `npm ci && npm run build`.
6. Test the plugin functionality.
7. Look for any 404 errors in console.
8. Look for any errors in the debug.log file.
9. Test the final ZIP.

### Changelog entry


> Dev - Update NPM packages and node version to v20 to modernize developer experience.
